### ISSUE TITLE
[chart/redis-ha][BUG] Can only do a rolling update of the ha-proxy deployment on a cluster with min 4 nodes

### ISSUE THREAD

**Issue by martijnvdp** [DandyDeveloper/charts#249]
**Describe the bug**
Can only do a rolling update of the ha-proxy deployment on a cluster with min 4 nodes
`k rollout restart deployment argocd-redis-ha-haproxy` 
the pod affinity rules and the default spec: 
```
    rollingUpdate:
      maxSurge: 25%
      maxUnavailable: 25%
```
requires a cluster of 4 nodes to work

changing the deployment spec to :
```
  strategy:
    type: RollingUpdate
    rollingUpdate:
      maxUnavailable: 1
```
fixes the issue

**To Reproduce**
start a rolling update of the haproxy deployment on a cluster with 3 nodes with afinnity rules set 
`hardAntiAffinity: true`

**Expected behavior**
i expected to be able to do a rolling update on a cluster with less than 4 nodes

**Additional context**
see also https://stackoverflow.com/questions/65063122/kubernetes-podantiaffinity-affects-deployment-failedscheduling-didnt-match

**Comment by martijnvdp** [comment]
argocd has this also added to the deployent helm chart
https://github.com/argoproj/argo-helm/blob/68f24d146c3c08e64117d98e3d58a6f073ec27a5/charts/argo-cd/values.yaml#LL141C17-L141C17

**Comment by DandyDeveloper** [comment]
Hello @martijnvdp 

I've been very neglectful of this repo as I've moved overseas back to my home country. 

I'll try to jump on your PR ASAP to get it in, I'll hopefully be settled over the coming week or two. 

Thanks for your patience.

**Comment by vlatitude** [comment]
Hello @DandyDeveloper, please take a look at the PR. It should be expected to perform a rolling upgrade using the default settings of 3 replicas

**Comment by DandyDeveloper** [comment]
Fixed in 4.25.1 per #250
